components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/PolicyManagerImpl.java

```diff
@@ -236,12 +236,14 @@ public boolean updatePolicyPriorities(List<Policy> policies) throws PolicyManage
     @Override
     public boolean deletePolicy(Policy policy) throws PolicyManagementException {
 
-        boolean bool;
         try {
             PolicyManagementDAOFactory.beginTransaction();
-            bool = policyDAO.deletePolicy(policy);
+            policyDAO.deleteAllPolicyRelatedConfigs(policy.getId());
+            policyDAO.deletePolicy(policy.getId());
+            featureDAO.deleteFeaturesOfProfile(policy.getProfileId());
+            profileDAO.deleteProfile(policy.getProfileId());
             PolicyManagementDAOFactory.commitTransaction();
-
+            return true;
```

Comment:
deletePolicy does not sound like returning a boolean. BTW, do we really need to return true here if transaction is successfull?